### DISCUSSION TITLE
Form inside form. When submitting the child form, the parent form will submit as well

### DISCUSSION THREAD

**Discussion by terror2012** [react-hook-form/react-hook-form#8889] [Q&A] [answered]
I currently have this type of hierarchy:

Base.tsx (has a formProvider and a form inside, and the other components inside both of them)
Form1.tsx (has the context of the base.tsx, and the modal child inside it)
ModalChild1.tsx (has a different form with a different onSubmit).

What I'm trying is pass the data from ModalChild1 to Form1 through prop function on form submit, but when I do submit it, both that form and the main form from base.tsx is submitted.

On ModalChild1 I'm using useForm() so it's not using the actual context.

Is there a way to submit only the modal form?

Thanks

**Comment by bluebill1049** [comment]
could you try to avoid form within form?

**Comment by leapful** [comment, marked as answer]
@terror2012 @bluebill1049 The issue might be related to this issue https://github.com/react-hook-form/react-hook-form/discussions/8896. Could you please check ?

**Reply by bluebill1049** [reply]
similar question, i am marking your comment as (redirect) answer

**Comment by Upinwv** [comment]
Canary Mail
You've received a secure email
***@***.*** has sent you a secure email via Canary Mail.

Read Secure Email (https://secure.canarymail.io/read?obj_id=fc4cf6ad-09be-48ad-8874-1c61e8e43912&obj_key=NENwU05MWm9NazAxTDR1RjM0Yk1adz09&thr_id=fc4cf6ad-09be-48ad-8874-1c61e8e43912)

If you expect to correspond often with ***@***.***, we recommend you download Canary Mail for free.

Download Canary (https://canarymail.io)

Privacy (https://canarymail.io/privacy.html) | Terms (https://canarymail.io/terms.html) | Docs (https://help.canarymail.io/) | Support (https://canarymail.zendesk.com/hc/en-us/requests/new)

Copyright © 2021 Canary Mail, All rights reserved.
